src/WithBackgroundImage/index.js

```diff
@@ -1,26 +1,70 @@
-import React from 'react';
+import React, {useEffect, useState} from 'react';
 import PropTypes from 'prop-types';
 
 import useBaseUrl from '@docusaurus/useBaseUrl';
-import useDocusaurusContext from '@docusaurus/useDocusaurusContext';
 import useThemeContext from '@theme/hooks/useThemeContext';
 
+let counter = 0;
+
+const getImage = (
+  [imageDark, imageDarkHover, imageLight, imageLightHover],
+  isHovered,
+) => {
+  if (!imageLight) {
+    return '';
+  }
+
+  const {isDarkTheme} = useThemeContext();
+
+  const backgroundImage = isDarkTheme && imageDark ? imageDark : imageLight;
+
+  let hoverBackgroundImage = isDarkTheme && imageDarkHover ? imageDarkHover : imageLightHover;
+  hoverBackgroundImage = hoverBackgroundImage ? hoverBackgroundImage : backgroundImage;
+
+  let image = isHovered ? hoverBackgroundImage : backgroundImage;
+
+  return useBaseUrl(image);
+};
+
+// TODO: Simplify the passing in of images
 const WithBackgroundImage = ({
   children,
+  className,
   imageDark,
-  imageLight, 
-  tag: Tag, 
+  imageDarkHover,
+  imageLight,
+  imageLightHover,
+  tag: Tag,
   ...props
 }) => {
-  const {isDarkTheme} = useThemeContext();
+  const [isHovered, setIsHovered] = useState(false);
+  const images = [imageDark, imageDarkHover, imageLight, imageLightHover];
 
-  const backgroundImage = isDarkTheme && imageDark ? imageDark : imageLight;
-  const backgroundImageStyle = backgroundImage 
-    ? { backgroundImage: `url('${useBaseUrl(backgroundImage)}')` } 
-    : {};
+  const image = getImage(images, isHovered);
+
+  const backgroundImageStyle = {
+    'backgroundImage': `url('${image}')`,
+  };
+  const imagesToPreload = images.filter(url => url).map(url => useBaseUrl(url));
```

Comment:
No need to change this now since the arrays small but we are looping over these too many times.

Reply:
I prefer it like this since as you said the size is small, so there's no affect on performance. IMO it's more readable this way